lib/src/error_tracking/dart_exception_processor.dart

```diff
@@ -0,0 +1,295 @@
+
+    // If it's an Error, try to use its built-in stackTrace
+    if (error is Error) {
+      effectiveStackTrace ??= error.stackTrace;
+    }
+
+    // If still null or empty, get current stack trace
+    if (effectiveStackTrace == null ||
+        effectiveStackTrace == StackTrace.empty) {
+      effectiveStackTrace = stackTraceProvider?.call() ?? StackTrace.current;
+      isGeneratedStackTrace = true; // Flag to remove top PostHog frames
+    }
+
+    // Check if we still have an empty stack trace
+    final hasValidStackTrace = effectiveStackTrace != StackTrace.empty;
+
+    // Process single exception for now
+    final frames = hasValidStackTrace
+        ? _parseStackTrace(
+            effectiveStackTrace,
+            inAppIncludes: inAppIncludes,
+            inAppExcludes: inAppExcludes,
+            inAppByDefault: inAppByDefault,
+            removeTopPostHogFrames: isGeneratedStackTrace,
+          )
+        : <Map<String, dynamic>>[];
+
+    final errorType = _getExceptionType(error);
+
+    // Mark exception as synthetic if:
+    // - runtimeType.toString() returned empty/null (fallback to 'Error' type)
+    // - Stack trace was generated by PostHog (not from original exception)
+    // - No valid stack trace is available
+    final isSynthetic =
+        errorType == null || isGeneratedStackTrace || !hasValidStackTrace;
+
+    final exceptionData = <String, dynamic>{
+      'type': errorType ?? 'Error',
+      'mechanism': {
+        'handled': true, // always true for now
+        'synthetic': isSynthetic,
+        'type': 'generic',
+      }
+    };
+
+    // Add exception message, if available
+    final errorMessage = error.toString();
+    if (errorMessage.isNotEmpty) {
+      exceptionData['value'] = errorMessage;
+    }
+
+    // Add stacktrace, if any frames are available
+    if (frames.isNotEmpty) {
+      exceptionData['stacktrace'] = {
+        'frames': frames,
+        'type': 'raw',
+      };
+    }
+
+    // Add thread ID, if available
+    final threadId = _getCurrentThreadId();
+    if (threadId != null) {
+      exceptionData['thread_id'] = threadId;
+    }
+
+    // Final result, merging system properties with user properties (user properties take precedence)
+    final result = <String, dynamic>{
+      '\$exception_level': 'error', // Never crashes, so always error
+      '\$exception_list': [exceptionData],
+      if (properties != null) ...properties,
+    };
+
+    return result;
+  }
+
+  /// Determines if a stack frame belongs to PostHog SDK (just check package for now)
+  static bool _isPostHogFrame(Frame frame) {
+    return frame.package == 'posthog_flutter';
+  }
+
+  /// Parses stack trace into PostHog format
+  static List<Map<String, dynamic>> _parseStackTrace(
+    StackTrace stackTrace, {
+    List<String>? inAppIncludes,
+    List<String>? inAppExcludes,
+    bool inAppByDefault = true,
+    bool removeTopPostHogFrames = false,
+  }) {
+    final chain = Chain.forTrace(stackTrace);
```

Comment:
check this out https://github.com/getsentry/sentry-dart/blob/a69a51fd1695dd93024be80a50ad05dd990b2b82/packages/dart/lib/src/sentry_stack_trace_factory.dart#L65C23-L69
stacktrace can be different types, so worth checking before doing `Chain.forTrace`

Reply:
There was a reason why it was done that way i just dont remember by heart, git blame might give you some hints